src/Framework/TestSuiteIterator.php

```diff
@@ -91,4 +123,72 @@ public function hasChildren(): bool
     {
         return $this->tests[$this->position] instanceof TestSuite;
     }
+
+    /**
+     * Reorder Tests within a TestCase in such a way as to resolve as many dependencies as possible.
+     * The algorithm will leave the tests in original running order when it can.
```

Comment:
What about cross-class dependencies? They are possible, though I rarely see them "in the wild".